maven-core/src/main/java/org/apache/maven/artifact/handler/manager/DefaultArtifactHandlerManager.java

```diff
@@ -53,6 +53,10 @@ public ArtifactHandler getArtifactHandler( String type )
             {
                 handler = new DefaultArtifactHandler( type );
             }
+            else
+            {
+                allHandlers.put( type, handler );
```

Comment:
What side effects may this cause if we don't receive a new instance?

Reply:
The handlers have no mutation methods, so I don't see what side effects this would cause. I'd say a stateful handler would be a terrible implementation.